lib/catalog/evaluate_order_process.rb

```diff
@@ -0,0 +1,86 @@
+module Catalog
+  class EvaluateOrderProcess
+    attr_reader :order
+
+    def initialize(task, order, tag_resources)
+      @task = task
+      @order = order
+      @tag_resources = tag_resources
+    end
+
+    def process
+      # TODO: Update this for when multiple applicable order items
+      # can be ordered.
+      @applicable_order_item = @order.order_items.first
+
+      relevant_order_processes = find_relevant_order_processes
+
+      # TODO: Put the order processes in order by their sequence number before
+      # doing this logic
+      before_sequence_number = 1
+      after_sequence_number = determine_starting_after_sequence_number(relevant_order_processes)
+
+      relevant_order_processes.each do |order_process|
+        if order_process.before_portfolio_item.present?
+          Api::V1x2::Catalog::AddToOrderViaOrderProcess.new(before_params(order_process, before_sequence_number)).process
+          before_sequence_number += 1
+        end
+
+        if order_process.after_portfolio_item.present?
+          Api::V1x2::Catalog::AddToOrderViaOrderProcess.new(after_params(order_process, after_sequence_number)).process
+          after_sequence_number -= 1
+        end
+      end
+
+      @applicable_order_item.update(:process_sequence => applicable_sequence(relevant_order_processes), :process_scope => "applicable")
+
+      self
+    end
+
+    private
+
+    def find_relevant_order_processes
+      tag_link_query = TagLink.where(:tag_name => all_tags)
+
+      OrderProcess.where(:id => tag_link_query.select(:order_process_id).distinct)
+    end
+
+    def all_tags
+      tag_pattern = '\/\w+\/order_processes=\d'
+
+      @tag_resources.map { |resource| resource[:tags] }.flatten.map { |tag| tag[:tag] }.select { |t| t.match?(tag_pattern) }.uniq
```

Comment:
Ok so I think I know the answer to this question after some digging, but I want to clarify:

So, we used to do `.(&:to_tag_string)` at the end of a similar line before in the old file, but it doesn't look like we're doing that anymore here. I know that the tags that are coming back from the `Tags::CollectLocalOrderResources` class are already getting the `to_tag_string` treatment, but I don't think that the ones from `Tags::Topology::RemoteInventory` are, at least not explicitly.

Are we sure we are getting back the same format across all of the tags? Or do we only need to do `to_tag_string` on the tags that *we* are storing in our db and not the `TopologicalInventoryApiClient::Tag` objects, because they use `.tag` instead? If that's the case, it seems like maybe we should mimic how that object works (not in this PR, but eventually)?

Reply:
Both `Tags::CollectLocalOrderResources` and `Tags::Topology::RemoteInventory` return the same format of hash named as `tag_resources`. Their `tags` attributes are already changed in the same format as of `to_tag_string`. 

https://github.com/RedHatInsights/catalog-api/blob/master/lib/tags/collect_local_order_resources.rb#L29
https://github.com/RedHatInsights/catalog-api/blob/master/lib/tags/topology/remote_inventory.rb#L24

And I agree that `to_tag_string` is only needed for the tags in our db, to change them to the same format as `TopologicalInventoryApiClient::Tag.tag`